### DISCUSSION TITLE
GitHub Actions Stuck and Slow to Cancel - Possible Bug?

### DISCUSSION THREAD

**Discussion by ar-ramchandra** [community/community#140130] [Actions]
### Select Topic Area

General

### Body

Hello GitHub Community and GitHub Team! 👋

First and foremost, I want to thank GitHub for its incredible platform and all the free tools and features it provides to developers like me. I truly appreciate the support GitHub offers to the developer community!

### The Issue

Recently, I encountered an issue with GitHub Actions where my workflow got stuck with the message:

After waiting for a while, I attempted to cancel the action multiple times from the repo's Actions tab. It took **45 minutes** for the cancel command to successfully terminate the job. While I'm glad it eventually worked, this delay is a bit concerning.

### My Request

I'm not raising this as a complaint; I just wanted to bring this to the community's and GitHub team's attention in case it's a potential bug or a rare occurrence that might need further investigation. If the GitHub team needs more details or wishes to investigate this further, I'm more than happy to provide any additional information. Please feel free to reach out to me!

### Additional Details

- **Workflow File**: `.github/workflows/main.yaml`
- **Job Configuration**: Uses `ubuntu-latest` runner.
- **Status**: The job got stuck in the "Waiting" state and eventually took 45 minutes to cancel after I initiated the cancel command.
- **Actions Taken**: Attempted to cancel the action manually, which eventually succeeded after 45 minutes.

### What I’d Like to Know

- Has anyone else experienced similar issues with stuck GitHub Actions or long cancellation times?
- Any suggestions for troubleshooting or possible workarounds if this happens again?

Thanks again to GitHub for its amazing platform, and I look forward to hearing any thoughts or suggestions from the community!

Best regards,  
Akshay

**Comment by ThisIs-Developer** [comment]
hey @Akshay-Ram-Ramchandra,

i understand your frustration! having a "waiting" state for 45 minutes before you could cancel it is definitely concerning.

many users have faced similar issues, like:

1. **Check Conditions**: avoid using `if: always()` in your workflow.
2. **Use GitHub CLI**: cancel with:
   ```bash
   gh run cancel <run_id>
   ```
3. **Dummy Commits**: making a dummy commit can help reset stuck workflows.
4. **Monitor GitHub Status**: check the [GitHub Status page](https://www.githubstatus.com/) for any issues.

hope this helps!

---
if it does, mark it as the answer! 😪

**Reply by ar-ramchandra** [reply]
Thank you so much for taking the time to post a reply.. I tried everything you suggested except for the GitHub CLI. Since the action is already stuck, I'm not able to try to see if it might fix it. Also, it didn't happen again.

**Reply by Capta1nRaj** [reply]
> Thank you so much for taking the time to post a reply.. I tried everything you suggested except for the GitHub CLI. Since the action is already stuck, I'm not able to try to see if it might fix it. Also, it didn't happen again.

How did you make him stop? Or after how much time it stopped?
